bag_tracker/src/main/scala/uk/ac/wellcome/platform/archive/bag_tracker/services/GetLatestBag.scala

```diff
@@ -0,0 +1,31 @@
+package uk.ac.wellcome.platform.archive.bag_tracker.services
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.Route
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.platform.archive.common.bagit.models.BagId
+import uk.ac.wellcome.platform.archive.common.storage.services.StorageManifestDao
+import uk.ac.wellcome.storage.NoVersionExistsError
+
+trait GetLatestBag extends Logging {
+  val storageManifestDao: StorageManifestDao
+
+  def getLatestBag(bagId: BagId): Route = {
+    storageManifestDao.getLatest(bagId) match {
```

Comment:
Fun fact: the current bags API makes two calls to DynamoDB:

* What's the latest version?
* What's the row for this version?

This will be slightly more efficient, and avoid weird edge cases where the latest bag changes between lookup and retrieval.